src/main/java/com/synopsys/integration/detect/lifecycle/boot/DetectBoot.java

```diff
@@ -207,19 +213,39 @@ public Optional<DetectBootResult> boot(String detectVersion, String detectBuildD
         } catch (DetectUserFriendlyException e) {
             return Optional.of(DetectBootResult.exception(e, propertyConfiguration, directoryManager, diagnosticSystem));
         }
+        AutoScanTypeDecider autoDetectTool = new AutoScanTypeDecider();
+        Map<DetectTool, Set<String>> scanTypeEvidenceMap = autoDetectTool.decide(hasImageOrTar, detectConfiguration);
+        logger.info("signature scan:{}", scanTypeEvidenceMap.containsKey(DetectTool.SIGNATURE_SCAN));
+        // This code to be moved to DetectRun.
+        BinaryScanBatch binaryScanBatch = new BinaryScanBatch();
+        CodeLocationNameGenerator codeLocationNameGenerator = detectConfigurationFactory.createCodeLocationOverride()
+            .map(CodeLocationNameGenerator::withOverride)
+            .orElse(CodeLocationNameGenerator.withPrefixSuffix("prefix", "suffix"));// Temporary code - discard after move to DetectRun.
+        CodeLocationNameManager codeLocationNameManager = new CodeLocationNameManager(codeLocationNameGenerator);
+        for (String path : scanTypeEvidenceMap.get(DetectTool.BINARY_SCAN)) {
+            File binaryScanFile = new File(path);
+            String codeLocationName = codeLocationNameManager.createBinaryScanCodeLocationName(
+                binaryScanFile,
+                binaryScanFile.getName(),// should be project name?
+                "1"//should be project version?
+            );
+            binaryScanBatch.addBinaryScan(new BinaryScan(new File(path), binaryScanFile.getName(), "1", codeLocationName));
```

Comment:
Even in Detect Run, it would be ideal to just call a method from AutonomousManager or any other class under the autonomous module here that simply provides the autonomously detected binaryScanBatch